### ISSUE TITLE
Bundle tree-sitter grammars and queries into the repository

### ISSUE THREAD

**Issue by igor-ramazanov** [phaazon/kak-tree-sitter#164]
First of all thank you for contribution of your work to the open source.

I am fairly new in Kakoune and do not know much about Tree Sitter yet,
but I noticed that comparing to Helix, it is a bit cumbersome to
use `ktsctl` in order to clone, compile and install tree sitter grammars and queries.

I think it would cool to bundle all grammars in the same way as Helix does (maybe we could can we even borrow their work on that).

So, after installing the `kak-tree-sitter` it would ready to go immediately.

Also, it would be cool, I think, to avoid cloning `Helix` multiple times, but do it only once.

The downside is depending on Helix and their list of supported languages,
although, we do not have to use Helix, the main idea is to precompile and bundle all interesting language into the repository.

Any thoughts?

**Comment by igor-ramazanov** [comment]
Ok, I've found your thoughts here: https://github.com/phaazon/kak-tree-sitter/issues/138#issuecomment-1740629635

**Comment by phaazon** [comment]
> Also, it would be cool, I think, to avoid cloning Helix multiple times, but do it only once.

I agree and that needs to be revamped indeed.

> The downside is depending on Helix and their list of supported languages,

Not completely, since I want to have a « URL cache » so that downloading twice from Helix doesn’t happen, but if someone wants to alter the config and depends on something different, they can.

> So, after installing the kak-tree-sitter it would ready to go immediately.

I’m not completely sure about that. I don’t think that’s what Helix does; I think it’s probably more a distribution thing. However, do go your way, I think it would be nice to have a single `ktsctl` command to install all the « currently available » grammars and queries, since we have a config file full of language information. It would require a bit of work (I want to have parallel cloning, parallel compiling, etc. etc.). If you’re interested about that, have a look at #30.

**Comment by igor-ramazanov** [comment]
Hey, I wrote a small nix package for myself with `kak-tree-sitter` which downloads and compiles Helix grammars and queries on installation.

It's been serving very well for me. Didn't have to use `ktsctl` since then.

**Comment by phaazon** [comment]
That’s a great news. You can clone the wiki repository and update the install instructions to include Nix; feel free to share a link to the branch so that I can add the remote and merge it.

**Comment by igor-ramazanov** [comment]
Oh nice :) Yeah may do that, will need to clean it up a bit
